main.rb

```diff
@@ -0,0 +1,95 @@
+module Enumerable
+
+    def my_each
+    unless !block_given?
+      for item in self
+        yield(item)
+      end
+    end
+      return self
+    end
+  
+    def my_each_with_index
+    unless !block_given?
+      for item_index in (0...self.size)
+        yield(self[item_index], item_index)
+      end
+    end
+      return self
+    end
+  
+    def my_select
+    unless !block_given?
+      new_items = []
+      my_each do |item|
+        new_items.push(item) if yield(item)
+      end
+      return new_items
+    end
+      return self
+    end
+  
+    def my_all?
+    return true if !block_given?
+    items_exist = true
+      my_each do |item|
+            items_exist = false if !yield(item)
+      end
+    items_exist
+    end
+  
+    def my_count
+      return self.size if !block_given?
+      count = 0
+      my_select do |item|
+          count+=1 if yield(item)
+      end
+      count
+    end
+  
+    def my_any?
+      result = false
+      my_each do |item|
+          result = true if yield(item)
+      end
+      result
+    end
+  
+    def my_none?
+      result = true
+      unless !block_given?
+        my_each do |item|
+            result = false if yield(item)
+        end
+      end
+      return result
+    end
+  
+    def my_map(&proc)
+      return self.size if !block_given?
+      new_array = []
+      my_select { |item| new_array << proc.call(item)}
+      new_array
+    end
+  
+
+    def my_inject initial=0
+        sum = initial
+        item_index = 0
+         
+        sum = self[0] if initial ==0
+        item_index =1 if initial ==0
+   
+        for item_index in (item_index...self.size)
+              sum = yield(sum, self[item_index])
+        end
+        sum
+      
+    end
```

Comment:
`my_inject` identical and returns the same thing as ruby's [`inject`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-inject). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_inject.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))
- combines all elements of enum by applying a binary operation, specified by a block
```ruby
[1, 2, 3].my_inject { |memo, num| memo + num } #should return 6
[1, 2, 3].my_inject(1) { |memo, num| memo + num } #should return 7
(1..3).my_inject(1) { |memo, num| memo + num } #should return 7
%w[dog door rod blade].my_inject {|memo, word| memo.length > word.length ? memo : word}  #should return 'blade'
```
- when a symbol is specified combines each element of the collection by applying the symbol as a named method
```ruby
[1, 2, 3].my_inject(:+)  #should return 6
[1, 2, 3].my_inject(1, :+)  #should return 7
(1.. 3).my_inject(1, :+)  #should return 7
```